Index: ghc/src/ghc-7.4.1/libraries/gen_contents_index
===================================================================
--- ghc.orig/src/ghc-7.4.1/libraries/gen_contents_index
+++ ghc/src/ghc-7.4.1/libraries/gen_contents_index
@@ -47,7 +47,6 @@ case $* in
 esac
 
 # Now create the combined contents and index pages
-echo $HADDOCK_ARGS
 $HADDOCK --gen-index --gen-contents -o . \
          -t "Haskell Hierarchical Libraries" \
          -p "prologue.txt" \
