--- Setup.hs.orig	2010-08-10 16:32:47.528967385 -0700
+++ Setup.hs	2010-08-10 16:32:50.063116088 -0700
@@ -51,7 +51,7 @@
            prelude_dst_dir = rgl_dst_dir lbi </> "prelude"
        createDirectoryIfMissing True prelude_dst_dir
        files <- getDirectoryContents prelude_src_dir
-       run_gfc pkg lbi (["-s", "--gfo-dir="++prelude_dst_dir] ++ [prelude_src_dir </> file | file <- files, take 1 file /= "."])
+       run_gfc pkg lbi (["+RTS", "-K20M", "-RTS", "-s", "--gfo-dir="++prelude_dst_dir] ++ [prelude_src_dir </> file | file <- files, take 1 file /= "."])
   , RGLCommand "lang"    True  $ \mode args pkg lbi -> do
        mapM_ (gfc mode pkg lbi . lang) (optl langsLang args)
        mapM_ (gfc mode pkg lbi . symbol) (optl langsAPI args)
@@ -297,7 +297,7 @@
 -- | Runs the gf executable in compile mode with the given arguments.
 run_gfc :: PackageDescription -> LocalBuildInfo -> [String] -> IO ()
 run_gfc pkg lbi args = 
-    do let args' = ["-batch","-gf-lib-path="++rgl_src_dir] ++ filter (not . null) args
+    do let args' = ["+RTS", "-K20M", "-RTS", "-batch","-gf-lib-path="++rgl_src_dir] ++ filter (not . null) args
            gf = default_gf pkg lbi
        putStrLn $ "Running: " ++ gf ++ " " ++ unwords (map showArg args')
        e <- rawSystem gf args'
